server/src/main/java/io/druid/server/QueryLifecycle.java

```diff
@@ -112,27 +114,36 @@ public QueryLifecycle(
    * either fully iterated or throws an exception.
    *
    * @param query             the query
-   * @param authorizationInfo authorization info from the request; or null if none is present. This must be non-null
-   *                          if security is enabled, or the request will be considered unauthorized.
+   * @param user              authentication token from the request
+   * @param namespace         authentication namespace of the request
    * @param remoteAddress     remote address, for logging; or null if unknown
+   * @param needsAuth         if false, skip the authorization check. This is useful when the authorization check has
+   *                          already been performed (e.g. in SQL handling, where authorization takes place in the
+   *                          planning step)
    *
    * @return results
    */
   @SuppressWarnings("unchecked")
   public <T> Sequence<T> runSimple(
       final Query<T> query,
-      @Nullable final AuthorizationInfo authorizationInfo,
-      @Nullable final String remoteAddress
+      @Nullable final String user,
+      @Nullable final String namespace,
+      @Nullable final String remoteAddress,
```

Comment:
The two existing callers of this method don't need the auth check here, but in general, if a caller does need auth, it would need to pass in those parameters.